sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +193,521 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
+			m.Notify(state.notify)
+			fakeTicker <- time.Now()
+			if len(state.expectedSensorMessage) > 0 {
+				select {
+				case <-time.After(10 * time.Second):
+					s.Fail("timeout")
+				case msg, ok := <-m.sensorUpdates:
+					s.Require().True(ok, "channel should not be closed")
+					s.Assert().NotNil(msg)
+				}
+			} else {
+				select {
+				case _, ok := <-m.sensorUpdates:
+					s.Require().True(ok, "channel should not be closed")
+					s.Fail("Should not received a message")
+				case <-time.After(time.Second):
+					break
+				}
+			}
+		})
+	}
+	m.Stop(nil)
+	m.done.Wait()
+}
+
+//endregion
+
+//region Helper functions
+
+func createManager(mockCtrl *gomock.Controller) (*networkFlowManager, *mocksManager.MockEntityStore, *mocksExternalSrc.MockStore, *mocksDetector.MockDetector) {
+	mockEntityStore := mocksManager.NewMockEntityStore(mockCtrl)
+	mockExternalStore := mocksExternalSrc.NewMockStore(mockCtrl)
+	mockDetector := mocksDetector.NewMockDetector(mockCtrl)
+	mgr := &networkFlowManager{
+		clusterEntities:   mockEntityStore,
+		externalSrcs:      mockExternalStore,
+		policyDetector:    mockDetector,
+		done:              concurrency.NewSignal(),
+		connectionsByHost: make(map[string]*hostConnections),
+		sensorUpdates:     make(chan *central.MsgFromSensor),
+		publicIPs:         newPublicIPsManager(),
+		centralReady:      concurrency.NewSignal(),
+	}
+	return mgr, mockEntityStore, mockExternalStore, mockDetector
+}
+
+type expectFn func()
 
-	suite.Len(h.endpoints, 1)
+func (f expectFn) runIfSet() {
+	if f != nil {
+		f()
+	}
 }
+
+func expectEntityLookupContainerHelper(mockEntityStore *mocksManager.MockEntityStore, times int, containerMetadata clusterentities.ContainerMetadata, found bool) expectFn {
+	return func() {
+		mockEntityStore.EXPECT().LookupByContainerID(gomock.Any()).Times(times).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
+			return containerMetadata, found
+		})
+	}
+}
+
+func expectEntityLookupEndpointHelper(mockEntityStore *mocksManager.MockEntityStore, times int, retVal []clusterentities.LookupResult) expectFn {
+	return func() {
+		mockEntityStore.EXPECT().LookupByEndpoint(gomock.Any()).Times(times).DoAndReturn(func(_ any) []clusterentities.LookupResult {
+			return retVal
+		})
+	}
+}
+
+func expectExternalLookupHelper(mockExternalStore *mocksExternalSrc.MockStore, times int, retVal *storage.NetworkEntityInfo) expectFn {
+	return func() {
+		mockExternalStore.EXPECT().LookupByNetwork(gomock.Any()).Times(times).DoAndReturn(func(_ any) *storage.NetworkEntityInfo {
+			return retVal
+		})
+	}
+}
+
+func expectDetectorHelper(mockDetector *mocksDetector.MockDetector, times int) expectFn {
+	return func() {
+		mockDetector.EXPECT().ProcessNetworkFlow(gomock.Any()).Times(times)
+	}
+}
+
+type connectionPair struct {
+	conn   *connection
+	status *connStatus
+}
+
+func createConnectionPair(incoming bool, external bool, invalid bool, firstSeen timestamp.MicroTS) *connectionPair {
```

Comment:
This may be an overkill here, but applying a builder pattern would make it easier to parse the 3 bools when reading the test cases.

Reply:
I went for it and I think it's way more readable :) Let me know what you think

Reply:
It looks good, thanks!